src/pages/studyView/charts/violinPlotTable/StudyViewViolinPlot.tsx

```diff
@@ -0,0 +1,193 @@
+import * as React from 'react';
+import { observer } from 'mobx-react';
+import {
+    ClinicalViolinPlotIndividualPoint,
+    ClinicalViolinPlotBoxData,
+} from 'cbioportal-ts-api-client';
+import { computed } from 'mobx';
+import {
+    getViolinX,
+    violinPlotSvgHeight,
+    violinPlotXPadding,
+} from 'pages/studyView/charts/violinPlotTable/StudyViewViolinPlotUtils';
+
+export interface IStudyViewViolinPlotProps {
+    curveMagnitudes: number[];
+    violinBounds: { min: number; max: number };
+    individualPoints: ClinicalViolinPlotIndividualPoint[];
+    boxData: ClinicalViolinPlotBoxData;
+    showViolin: boolean;
+    showBox: boolean;
+    width: number;
+    gridValues: number[];
+}
+
+export const yPadding = 3;
+export const height = violinPlotSvgHeight - 2 * yPadding;
+
+@observer
+export default class StudyViewViolinPlot extends React.Component<
+    IStudyViewViolinPlotProps,
+    {}
+> {
+    @computed get svgWidth() {
+        return this.props.width;
+    }
+    @computed get violinWidth() {
+        return this.svgWidth - 2 * violinPlotXPadding;
+    }
+    renderCurve() {
+        if (this.props.curveMagnitudes.length > 0) {
+            const center = violinPlotSvgHeight / 2;
+
+            const curveLimit = height / 2;
+            const scale = Math.min(
+                1,
+                curveLimit / Math.max(...this.props.curveMagnitudes)
+            );
+            const curve = this.props.curveMagnitudes.map(y => scale * y);
+
+            const stepSize = this.violinWidth / (curve.length - 1);
+
+            let dUpperHalf = `M 0 ${center} L 0 ${center - curve[0]}`;
+            let dLowerHalf = `M 0 ${center} L 0 ${center + curve[0]}`;
+            for (let i = 1; i < curve.length; i++) {
+                dUpperHalf = `${dUpperHalf} L ${i * stepSize} ${center -
+                    curve[i]}`;
+                dLowerHalf = `${dLowerHalf} L ${i * stepSize} ${center +
+                    curve[i]}`;
+            }
+            dUpperHalf = `${dUpperHalf} L ${this.violinWidth} ${center} L 0 ${center}`;
+            dLowerHalf = `${dLowerHalf} L ${this.violinWidth} ${center} L 0 ${center}`;
+
+            return (
+                <g transform={`translate(${violinPlotXPadding}, 0)`}>
+                    <path d={dUpperHalf} fill={'gray'} />
+                    <path d={dLowerHalf} fill={'gray'} />
+                </g>
+            );
+        } else {
+            return null;
+        }
+    }
+
+    private x(v: number) {
+        return getViolinX(v, this.props.violinBounds, this.violinWidth);
+    }
+
+    renderPoints() {
```

Comment:
what are the individual points in a violin chart?  outliers or something like that?